src/transformers/generation_utils.py

```diff
@@ -1230,6 +1230,13 @@ def generate(
                 inputs_tensor, pad_token_id, eos_token_id
             )
 
+        # decoder-only models should use left-padding for generation
+        if not self.config.is_encoder_decoder and torch.sum(inputs_tensor[:, -1] == pad_token_id) > 0:
```

Comment:
Looks like `inputs_tensor` is not always a tensor from the tests??

Reply:
it was not the cause -- when `pad_token_id` is `None`,  `inputs_tensor[:, -1] == pad_token_id` doesn't evaluate to a tensor :D